File: Java_Joc_Windows_Android/AndroidStudioProject/core/src/base/menu/util/MenuItemConfig_SmartItem.java
package base.menu.util;

import java.lang.reflect.Constructor;

import base.visual.util.VisualPropertiesConfig;

public class MenuItemConfig_SmartItem extends MenuItemConfig implements IMenuItemConfig_DynamicProperties {

    public String[] parentVisualID;

    public String initialBaseStateId;

    public Constructor classContructor;

	private VisualPropertiesConfig itemProperties;

	public MenuItemConfig_SmartItem(String itemID, String[] parentVisualID, String initialBaseStateId, Constructor classContructor, VisualPropertiesConfig itemProperties) {
		super(itemID);

        this.parentVisualID = parentVisualID;

        this.initialBaseStateId = initialBaseStateId;

        this.classContructor = classContructor;

		this.itemProperties = itemProperties;
	}

	@Override
	public VisualPropertiesConfig getItemProperties() {
		return itemProperties;
	}

}

File: CPP_Joc_Windows_Android/SH_Client_Win_Cpp_Cmake/App/src/base/menu/util/MenuItemConfig_SmartItem.cpp
#include "MenuItemConfig_SmartItem.h"

MenuItemConfig_SmartItem::MenuItemConfig_SmartItem(
    std::string itemID,
    UiId* parentVisualID,
    std::string initialBaseStateId,
	std::function<ComposedStiledVisual*(IAppMetrics* appMetrics, IAppAssets* appAssets)> classContructor,
    VisualPropertiesConfig* itemProperties)
    : super(itemID),
      parentVisualID(parentVisualID), initialBaseStateId(initialBaseStateId), classContructor(classContructor), itemProperties(itemProperties)
{
    //void
}

MenuItemConfig_SmartItem::MenuItemConfig_SmartItem(
	std::string itemID,
	std::string parentComposedID,
	std::string initialBaseStateId,
	std::function<ComposedStiledVisual* (IAppMetrics* appMetrics, IAppAssets* appAssets)> classContructor,
	VisualPropertiesConfig* itemProperties)
    : MenuItemConfig_SmartItem(
		itemID,
		UiId::newIdFromComposedId(parentComposedID, "_local"),
		initialBaseStateId,
		classContructor,
		itemProperties)
{
    //void
}

VisualPropertiesConfig* MenuItemConfig_SmartItem::getItemProperties() {
    return itemProperties;
}

MenuItemConfig_SmartItem::~MenuItemConfig_SmartItem() {
    //void
}
